src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -308,6 +316,11 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
                 // See http://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Security
                 if (!string.IsNullOrEmpty(message.IdToken) || !string.IsNullOrEmpty(message.AccessToken))
                 {
+                    if (Options.ShareCallbackPath)
```

Comment:
This extra check is here to ensure authorization responses using an unsafe `response_mode`/`response_type` combination (i.e an identity or an access token flowed using the query string) are always rejected, as it's necessarily the result of a non-standard OIDC server doing crazy things (personally, I'd even be tempted to add a check for `refresh_token`).

Since it's a serious protocol exception, I'm not sure skipping the request is the right thing to do.